### DISCUSSION TITLE
Experience with alternative embeddings?

### DISCUSSION THREAD

**Discussion by philippbayer** [Future-House/paper-qa#269] [Q&A]
Hi! I have several hundred to thousands of PDFs and I'd rather not run them all through openAI's embedding API as the costs will pile up. I'm sure that using free, alternative embeddings such as SentenceTransformers will have an impact on performance. Have you made any experience with combinations of 'open' embeddings combined with closed openAI bots in terms of quality of answers?

**Comment by Snikch63200** [comment]
I use an hybrid embedding model that includes mixbread-enbed-large, as follows : 

```
    sparse_embedding_model = SparseEmbeddingModel(ndim=2048)
    sentence_transformer = SentenceTransformerEmbeddingModel(name="mixedbread-ai/mxbai-embed-large-v1",
                                                             config={"device": "cuda"}  # Set the device to GPU
                                                             )
    hybrid_model=HybridEmbeddingModel(models=[sparse_embedding_model,sentence_transformer])
```

Gives good results.

Best regards
